### ISSUE TITLE
bug(android): Install Updates option is missing in Keyman menu list after setting the device to current date and time

### ISSUE THREAD

**Issue by bharanidharanj** [keymanapp/keyman#8557]
### Describe the bug

I noticed that while trying to upgrade the sil_cameroon_qwerty keyboard version from 6.0.4 to 6.0.8, the Install updates option is missing on the Keyman menu list. 

### Reproduce the bug

1. Click the Keyman App from an Android device. 
2. Verify that the Keyman opens successfully
3. Verify that the ‘Get Started’ dialog opens successfully. 
4. Click the close button from the Get Started dialog. 
5. Verify that the Get Started dialog closes
6. Click the three dots button which appears at the top right corner of the text pane. 
7. Click the Settings option. 
8. Verify that the Settings page opens. 
9. Click the ‘Install Keyboard or Dictionary’ option. 
10. Verify that the ‘Install Keyboard or Dictionary’ page opens. 
11. Click the ‘Install from local file ‘ option. 
12. Verify that it opens the ‘Downloads’ folder. 
13. Click  sil_cameroon_qwerty.kmp file (an old version 6.0.4) 
14. Verify that the Cameroon Qwerty keyboard has been successfully installed. 
15. Click the Back button from the Keyman app. 
16. Verify that the Keyman stops running on the device. 
17. Click Settings on the Android device. 
18. Click Display options. 
19. Set the date to automatically update from the internet. 
20. Turn off Airplane mode .
21. Verify it turns on internet access. 
22. Click Display options. 
23. Click the Keyman App from an Android device. 
24. Verify that the Keyman opens successfully
25. Verify that the ‘Get Started’ dialog opens successfully. 
26. Click the close button from the Get Started dialog. 
27. Verify that the Get Started dialog closes
28. Click the three dots button which appears at the top right corner of the text pane. 
29. Wait for the catalog to update (for few minutes)
30. Click on the Keyman Settings menu (three dots) option. 

Here, I noticed that the Install Updates option is missing from the Keyman menu list. Seems to be an issue. 
  
   <img width="300" src="https://user-images.githubusercontent.com/19683143/229128719-eb442673-7034-4534-82d8-a5f58c93bec1.PNG">

### Expected behavior

Install Updates option should be in Keyman menu list after updating Date and time settings. 

### Related issues

https://github.com/keymanapp/keyman/issues/6731

### Keyman apps

- [X] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.80-alpha

### Operating system

Android 5.0, 9.0, 12.0

### Device

API 21, 28 and 31  Android Emulator

**Comment by darcywong00** [comment]
imho, I'd rather remove the existing date-checking and implement automatic keyboard updates in #7171.

**Comment by mcdurdin** [comment]
Yes, but we'd still need the date-checking to schedule updates, and it'd be important to have a pathway to trigger a manual update check.

**Comment by darcywong00** [comment]
Per Sprint A17S11 sprint plan triage, this will defer to Future and be implemented as a menu option to force a check on keyboard/lexical-models updates

**Comment by bharanidharanj** [comment]
@darcywong00 @mcdurdin Able to reproduce this issue with the latest Keyman 17.0.205-alpha build in Android 5.0 lollipop / API 21 emulator.
